coderd/database/dbauthz/dbauthz.go

```diff
@@ -1195,17 +1195,39 @@ func (q *querier) GetTemplateDAUs(ctx context.Context, arg database.GetTemplateD
 }
 
 func (q *querier) GetTemplateDailyInsights(ctx context.Context, arg database.GetTemplateDailyInsightsParams) ([]database.GetTemplateDailyInsightsRow, error) {
-	// FIXME: this should maybe be READ rbac.ResourceTemplate or it's own resource.
-	if err := q.authorizeContext(ctx, rbac.ActionRead, rbac.ResourceSystem); err != nil {
-		return nil, err
+	for _, templateID := range arg.TemplateIDs {
+		template, err := q.db.GetTemplateByID(ctx, templateID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := q.authorizeContext(ctx, rbac.ActionUpdate, template); err != nil {
+			return nil, err
+		}
+	}
+	if len(arg.TemplateIDs) == 0 {
+		if err := q.authorizeContext(ctx, rbac.ActionUpdate, rbac.ResourceTemplate.All()); err != nil {
+			return nil, err
+		}
 	}
```

Comment:
This doesn't align with our documented permissions [here](https://coder.com/docs/v2/latest/admin/users#roles).

Is the reason for making the required permission "update" to ensure that only template owners can view template insights? If so, we should make a separate `TemplateInsights` resource and make appropriate modifications to our built-in roles.

Otherwise, why not just allow anyone with read permissions to view this data?

Reply:
> Is the reason for making the required permission "update" to ensure that only template owners can view template insights? 

> Otherwise, why not just allow anyone with read permissions to view this data?

I suppose that we don't want to share user details for template usage and latency with regular users.

> This doesn't align with our documented permissions [here](https://coder.com/docs/v2/latest/admin/users#roles).

The Roles table does not describe the use case when a **regular user** can be granted via Template ACLs to become a template admin (enterprise feature). This is why I picked the for-loop with "update template" check to see if the user can modify a template, ergo can access insights.

Please let me know what is your recommendation in this case.

Reply:
For the moment, this is OK as a bandage.